### DISCUSSION TITLE
"Save File Couldn't Be Resigned" Error

### DISCUSSION THREAD

**Discussion by dw1260** [bucanero/apollo-ps3#27] [Q&A] [answered]
Hi, I'm not sure what's causing my error message and I'd like to see what I can do on my end to resolve it, but I'm not sure where to start. Apollo opens up fine on my ps3 and appears to run normally, but I'm not sure if I have to add additional data about my system to a config file to get it to resign (since the Saved Data Owner is listed as Saved User but I don't recall entering that information anywhere). Apologies if this is remedial or solved in a trivial fashion, I'm just not quite understanding where the error is coming from.

I'm running a CFW ps3, 4.86 COBRA enabled, downloaded Apollo through brewology. If there's further information I need to submit to get to the issue, please let me know. I can't Brute Force anything or use other PC apps to resign, as I'm running a Mac and they won't cooperate with my Wine setup, so I'm trying to do everything through Apollo installed on the ps3 itself. Thanks for any help you can provide.

**Comment by bucanero** [comment, marked as answer]
the most probable issue is that the game you're trying to resign is not listed in the secure file id database.
Without the encryption key for right the Title_ID, Apollo won't be able to resign the file.

You can check here, if your TITLE_ID is listed or not:
https://github.com/bucanero/apollo-ps3/blob/master/appdata/games.conf

btw, in the "About" menu you should see a box with the detected information about your PS3. You can also validate that the detection was successful. 

Let me know if the savegame title was in the database or not.

**Reply by successtheman** [reply]
Out of general curiosity, is it possible to manually edit this file on the ps3 itself? or is this stored serverside?

**Reply by bucanero** [reply]
as mentioned, the file can be edited/replaced since it's stored on the PS3 HDD: `/dev_hdd0/game/NP0APOLLO/USRDIR/DATA/games.conf`

**Comment by dw1260** [comment]
Ah, there you go - it's not there. I must have missed that part at some point. Is there a way for me to add it? I'm not super familiar with the process, so again, sorry if this is kinda remedial. The "About" menu has my user and account ID displayed just fine btw, so it looks like the only issue is the database. If I can dump anything or add to it to contribute just let me know.

**Comment by bucanero** [comment]
some firmwares and jailbreaks allow to capture stuff like the secure file ids, game disc keys, etc.

An easy way to get the key, if you're using HEN (or a CFW like Rebug)
- check the XMB's `Hybrid firmware tools` menu, in the "Dump" section there's an option to dump Secure File Ids. 
- Enable the option, 
- then run the game and load a savegame. 
- The secure key will be saved in `/dev_hdd0/tmp/cfw_settings.log`

you can update the `games.conf` file with the right entry to be able to sign your game.

**Reply by abdogm** [reply]
> some firmwares and jailbreaks allow to capture stuff like the secure file ids, game disc keys, etc.
> 
> An easy way to get the key, if you're using HEN (or a CFW like Rebug)
> 
> * check the XMB's `Hybrid firmware tools` menu, in the "Dump" section there's an option to dump Secure File Ids.
> * Enable the option,
> * then run the game and load a savegame.
> * The secure key will be saved in `/dev_hdd0/tmp/cfw_settings.log`
> 
> you can update the `games.conf` file with the right entry to be able to sign your game.

Where i can find `games.conf` file ?

**Reply by bucanero** [reply]
you can find it here: `/dev_hdd0/game/NP0APOLLO/USRDIR/DATA`

if you have any additional secure file key that is missing from the list, please share it here so I can add the key to the next release.

**Reply by abdogm** [reply]
> you can find it here: `/dev_hdd0/game/NP0APOLLO/USRDIR/DATA`
> 
> if you have any additional secure file key that is missing from the list, please share it here so I can add the key to the next release.

Here is one.
```
;"J-STARS Victory VS+"
[NPUB31647/NPEB02204] 
;disc_hash_key= 
secure_file_id:*=01040204010101010101010101010101
```
BTW, i did what you said about adding key to `game.conf` and i got massage tells me the save was "successfully modified" but the game still saying the save file corrupted i tried other downloaded save files and i resign it and the game tells me the same thing save file is corrupted anything to solve this problem and am sorry if i being  annoying still new for ps3 jailbroken stuff 
 ¯\_(ツ)_/¯

**Reply by bucanero** [reply]
have you tried with any other save-game file from other games? for a quick test, just take any other game, and save. Then go to Apollo, and just "resign" that save, then go back to the game and try to load it. 
If it fails, then there's something broken with the setup, and the resign is failing for any game. If it works, then it's something about that J_stars game.

**Reply by CeruleanSnake** [reply]
> you can find it here: `/dev_hdd0/game/NP0APOLLO/USRDIR/DATA`
> 
> if you have any additional secure file key that is missing from the list, please share it here so I can add the key to the next release.

Apologies for necroposting this but google sent me here when I was having the same issue with my Band Hero save and these steps solved it for me (Was trying to move my very ancient save from my ps3 slim to rpcs3 for preservation).

I figured id share the key so you can throw it into the list.

```
; "Band Hero"
[BLUS30372/BLES00711]
;disc_hash_key=
secure_file_id:*=5F5F58424F5833363020465457215F5F
```

**Reply by bucanero** [reply]
> 
> Apologies for necroposting this but google sent me here when I was having the same issue with my Band Hero save and these steps solved it for me (Was trying to move my very ancient save from my ps3 slim to rpcs3 for preservation).
> 
> I figured id share the key so you can throw it into the list.
> 
> ```
> ; "Band Hero"
> [BLUS30372/BLES00711]
> ;disc_hash_key=
> secure_file_id:*=5F5F58424F5833363020465457215F5F
> ```

Thanks for sharing, this game key has been added to the key database. 👍

**Comment by dw1260** [comment]
Okay, I updated the config file with the right game information, and Apollo is recognizing the save file and giving me the "successfully modified" message, but the game's still not recognizing it - I'm noticing that when I remove the account ID, nothing's replacing it. It's just blank where my Account ID would be in my other save files, and it still reads "Not Owner." Am I supposed to input my Account ID manually somewhere, or should it be doing that when I resign the file? I'm not sure if this is a setting I should be changing within Apollo itself or somewhere else. I appreciate the feedback and help so far btw.

**Comment by bucanero** [comment]
There's no need to select the `remove Account ID` option (unless you actually want to get a blank Account ID in your savegame 😄  )

So if you just want to resign a savegame to use it in your PS3, simply select the savegame and click on `Apply changes and Resign` , without selecting any other option. That should be enough to resign a save to the current user and PS3 console.
After that, you can check the savegame details and it should show the proper account ID.

also, if you want to be extra-sure, I suggest doing all the changes on a USB (fat32) drive, and then copy the save to the HDD using the ps3 XMB. You'll get an additional bonus, because XMB validates the signatures and data so if it gets copied, then the resigning is valid. If there's a problem, XMB will not copy and report "savegame corrupted".

**Comment by dw1260** [comment]
Ah, I see... so when I apply changes and resign, it still lists the original Account ID with (Not Owner) next to it. That's why I started messing with the remove function in the first place. It has "Saved User" listed under game data owner in settings, but it's not actually changing anything when I resign it. Do I need to manually input my Account ID somewhere?

**Comment by bucanero** [comment]
the account ID that Apollo uses for resign, is the one that was detected and is shown in the "About" screen.

if you want to manually force values, see the Readme about the `owners.xml` , but as I said, the detected values should work unless you're doing some weird spoofing with your PS3 setup.

**Comment by dw1260** [comment]
Okay, so it turns out the Owner ID just wasn't appearing to change until I backed out and refreshed the save game list, then it was reflecting the change. The copying over from USB method seems to be the only way to get my PS3 to recognize the save file, loading it directly into the HDD (via FTP) would show the save in Apollo but not in the XMB, but, basically on Apollo's end everything is working. The save file itself isn't reflecting correctly in my game when I try and load it, but I'm not sure that's a problem with Apollo vs. a problem with the contents of the save file I'm trying to use. I'll have to look into that, but as far as Apollo goes it seems to be working just fine. Thank you for helping me clear that up.

**Comment by bucanero** [comment]
keep in mind, if you transfer a save directly to the HDD using FTP it won't show up on the XMB. The PS3 uses an index database so forcing files won't make them show up. (unless you boot in fail-safe mode and select `rebuild database`)

so the USB way is recommended, because it's clean and doesn't require the DB rebuild.

Note: you can do a direct copy to the HDD if you already have a savegame and you actually overwrite the data... then the database won't see the difference.

for specific game issues, I'd suggest checking forums such as [psx-place.com](http://www.psx-place.com), there are many PS users that might be able to help you. (and many of them use my homebrew tools already)

**Comment by angrynewraze** [comment]
I can't resign my rpcs3 tales of xillia save. I updated the games.conf with

;“Tales of Xillia”
[BLJS10120/BLES01815/BLUS31006/BLUS41029/NPUB31136]
;disc_hash_key=
secure_file_id:*=2E7F0BC9F5383AB3DC4E8EA303F04D8A

But it's still not letting me resign my save file or add the cheats.

**Comment by bucanero** [comment]
> I can't resign my rpcs3 tales of xillia save. I updated the games.conf with

RPCS3 is not supported, the emulator does not use the original PS3 save format.

https://github.com/bucanero/apollo-ps3/discussions/28#discussioncomment-219046

**Reply by WilletsM4** [reply]
I'm having a similar issue I started a thread can you help me out

**Reply by angrynewraze** [reply]
Bruteforce sometimes doesn't work with the emulator and Apollo definitely doesn't work with the emulator. I ended up having to buy a jailbroken ps3 to use cheats with my ps3 games.

**Reply by WilletsM4** [reply]
i'm on a cfw PS3 and having the same issue
